sylt-parser/src/parser.rs

```diff
@@ -797,7 +803,67 @@ fn assignable_index<'t>(ctx: Context<'t>, indexed: Assignable) -> ParseResult<'t
     sub_assignable(ctx, result)
 }
 
-/// Parse an [AssignableKind::Access].
+/// Parse an [AssignableKind::Access] or [AssignableKind::Variant].
+fn assignable_dot_or_variant<'t>(
+    ctx: Context<'t>,
+    accessed: Assignable,
+) -> ParseResult<'t, Assignable> {
+    // TODO(ed): It might be possible to remove this branch?
+    // TODO(ed): We throw away error information here...
+    match assignable_variant(ctx, accessed.clone()) {
+        Ok(variant) => Ok(variant),
+        Err(_) => assignable_dot(ctx, accessed),
+    }
+}
+
+/// Parse an [AssignableKind::Variant].
+fn assignable_variant<'t>(ctx: Context<'t>, accessed: Assignable) -> ParseResult<'t, Assignable> {
+    let span = ctx.span();
+    // TODO(ed): I kinda dislike this - but I think it's okay?
```

Comment:
I feel like it is not needed? Blobs are kind of like this already, so maybe we can replace that implementation with enums when we have anonymous blobs...

Reply:
?
The comment is about picking apart what was previous to see if this is a valid Variant. I'm not sure what you're talking about.